### ISSUE TITLE
System Navigation not working when launcher taskbar is disabled

### ISSUE THREAD

**Issue by hazzl** [crdroidandroid/issue_tracker#244]
### Issue type

User interface

### Device

OnePlus 6 (enchilada)

### crDroid version

crDroid 9

### Exact version / Build date

9.7/2023-07-29

### Bug description

If I disable the taskbar in crDroid-home (the standard launcher), I can no longer use back navigation with gesture navigation (swipe from the sides). The gesture is no longer recognised. Home navigation gesture (swipe from bottom) continues to work. If I try enabling 2- or 3-button navigation, the navigation keys don't show.

### Steps to reproduce

Step 1: use crDroid-home as your launcher and enable gesture system navigation.
Step 2: long press on the home screen to open home settings
Step 3: go to misc settings and disable the taskbar
Step 4: restart the launcher by going back to the home screen
Step 5: swipe from the edge of the screen: back gesture no longer triggers

### Relevant log (logcat/build log)

```shell
No log
```

### Acknowledgements

- [X] I've checked device is officially supported (for device specific reports and not source related).
- [X] I'm running latest version available on crdroid.net for this device
- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by hazzl** [comment]
I have confirmed, that this issue is still present in 9.8/2023-09-04. Maybe related to #228 ?

**Comment by gwolf2u** [comment]
do please check with crDroid 9.10

**Comment by hazzl** [comment]
Sorry, 9.10 isn't available yet for my device. However, the bug is still present in 9.9/2023-10-16

**Comment by gwolf2u** [comment]
Do update when crDroid 9.10 ready for your device
I know that @Terminator-J is working on it

**Comment by hazzl** [comment]
I'll update as soon as he releases the first 9.10-test build.

**Comment by Terminator-J** [comment]
I can't say I can reproduce. I never use that taskbar (always seemed buggy), but I tried enabling it.
Then I disabled it again.
Back gesture worked the entire time and still does.

However, if you go into Settings > crDroid Settings > Navigation, and toggle off the "Navigation bar" toggle, that will break gesture navigation as the system relies on that to "catch" other gestures.

I'm assuming that's not what you're talking about, since you were very specific.

Don't know when the feature broke, since like I said, I never used the "taskbar" launcher thing you're referring to, but in the current 9.10 sources (just re-synced this morning in case), back gestures seem to continue working when turning it back off.

 
> If I disable the taskbar in crDroid-home (the standard launcher), I can no longer use back navigation with gesture navigation (swipe from the sides). The gesture is no longer recognised. Home navigation gesture (swipe from bottom) continues to work. If I try enabling 2- or 3-button navigation, the navigation keys don't show.
> ### Steps to reproduce
> 
> Step 1: use crDroid-home as your launcher and enable gesture system navigation. Step 2: long press on the home screen to open home settings Step 3: go to misc settings and disable the taskbar Step 4: restart the launcher by going back to the home screen Step 5: swipe from the edge of the screen: back gesture no longer triggers

**Comment by Terminator-J** [comment]
If you want to check out test releases @hazzl they're pinned in the Telegram group, or you can look at:
https://github.com/TJ-Releases/enchilada/releases/tag/cr9.10_test1
https://github.com/TJ-Releases/fajita/releases/tag/cr9.10_test1

**Comment by hazzl** [comment]
> However, if you go into Settings > crDroid Settings > Navigation, and toggle off the "Navigation bar" toggle, that will break gesture navigation as the system relies on that to "catch" other gestures.

Oh that fixed it!

However, I think the configuration setting is not worded very well. It reads "Navigation bar: Display navigation bar at bottom of screen" and I specifically did not want to see the navigation bar, so I disabled that setting. After enabling the "Navigation bar" toggle, I still don't see the navigation bar (as I have enabled Gesture Navigation), so I'm actually not sure what that toggle does. For me it's a "unbreak gesture navigation" setting.
